MatrixKit/Models/Room/MXKRoomDataSourceManager.m

```diff
@@ -140,6 +140,27 @@ + (NSUInteger)missedDiscussionsCount
     return roomCount;
 }
 
++ (void)markAllMessagesAsRead
+{
+    // Reset the unread count in all the existing room data sources.
+    @synchronized(_roomDataSourceManagers)
+    {
+        NSArray *mxSessionIds = _roomDataSourceManagers.allKeys;
+        for (NSString *mxSessionId in mxSessionIds)
+        {
+            MXKRoomDataSourceManager *roomDataSourceManager = [_roomDataSourceManagers objectForKey:mxSessionId];
+            if (roomDataSourceManager)
+            {
+                NSArray *roomDataSources = roomDataSourceManager->roomDataSources.allValues;
+                for (MXKRoomDataSource *roomDataSource in roomDataSources)
+                {
+                    [roomDataSource markAllAsRead];
```

Comment:
This scares me.
It can produce one request to the HS per room, ie hundreds of request in parallel